src/persist/src/s3.rs

```diff
@@ -100,9 +100,9 @@ impl S3BlobConfig {
                 // maximum time allowed for a single network call
                 .operation_attempt_timeout(Duration::from_secs(90))
                 // maximum time until a connection succeeds
-                .connect_timeout(Duration::from_secs(30))
+                .connect_timeout(Duration::from_secs(2))
```

Comment:
I wouldn't mind slightly higher on this one, since we don't know our typical connect timings right now. 2s is probably fine, but I'd feel a little more comfortable bumping to maybe 3-4s until we have more metric coverage

Reply:
This might be wrong, but the way I read [the docs](https://docs.rs/aws-config/0.53.0/aws_config/timeout/struct.TimeoutConfig.html#method.read_timeout) the `read_timeout` includes the time required to connect, so the read timeout should always be greater than the  connect timeout.

So if we want to bump the connect timeout to 4s, and still allow for 2s to read something from the connection, we should set the read timeout to 6s. Does that make sense?

Btw, these values are total guesswork. Happy to increase them until we all feel comfortable!

Reply:
maybe 3 for connect and 6 for read then? that'd make me a little more comfortable when first rolling this out. I have a PR incoming that'll add some metric coverage + make the timeouts dynamic (once LD is hooked in), so we can lower them further more safely afterwards